### ISSUE TITLE
Code comment to opt out of linting for section of code.

### ISSUE THREAD

**Issue by bjankord** [stylelint/stylelint#105]
Just found about this project, it looks great! Haven't read through all the docs so not sure if this exists already, but I'd like the ability to opt-out of linting a section of code by adding comments in my stylesheet similar to [eslint](http://eslint.org/docs/user-guide/configuring.html#configuring-rules).

Something like

````css
.foo {
 color: #f00;
}

/* stylelint-disable */
#bar {color: #0f0 !important;}
/* stylelint-enable */
```

**Comment by davidtheclark** [comment]
Yep, that'd definitely be a nice feature that have.

**Comment by MoOx** [comment]
Sure we need that. We should even allow to enable/disable some rules

```
/* stylelint-disable whatver-rule */
```

**Comment by davidtheclark** [comment]
I'm thinking right now that one way to do this would be the following:

Before any rules are checked, cycle through all the comments and stash the line numbers that start and end these disabled-ranges -- maybe create an array of "disabling" objects like this:

```js
[{
  start: 3,
  end: 10,
  rules: ["whatever-rule"]
}]
```

(I think we'd want to ensure that these disabled-ranges do not overlap, otherwise things could get weird. User could only write a new `disable` comment *after* the previous disabled-range has been closed with an `enable`. Is that too rigid?)

Then when a rule is checked, if a warning is going to be registered because an error has been found, *first* ensure that the line number of the error is not within one of these disabling-ranges whose rule-list includes the error's rule.

Does that seem like the right approach, or does anyone know of a better way?

**Comment by MoOx** [comment]
Lgtm

**Comment by jeddy3** [comment]
> (I think we'd want to ensure that these disabled-ranges do not overlap, otherwise things could get weird. User could only write a new disable comment after the previous disabled-range has been closed with an enable. Is that too rigid?)

I don't think it'll be too rigid. Either way, it feels like the right way to start, with the additonally complexity only being added later down the line if needed.

> Then when a rule is checked, if a warning is going to be registered because an error has been found, first ensure that the line number of the error is not within one of these disabling-ranges whose rule-list includes the error's rule.

So, just to clarify, is this happening inside the rule or within `src/index.js`? The latter makes more sense to me e.g. each rule just detects and throws warnings, and then afterwards `src/index.js` is responsible for purging the warnings array of any warnings that fall between disabling comments. I think this approach will help keep the rules laser-focused and easier to code as the responsibility for managing the disable comments is passed to the `src/index.js`. Is that what you were thinking?

**Comment by davidtheclark** [comment]
We might want to create a wrapper for `result.warn()` --- that would allow us to run the warning through whatever additional processing we want and also in the future adjust to potential postcss API changes. Any opposition to that?

I'm going to assign this to myself and give it a try at some point today.

**Comment by jeddy3** [comment]
> Any opposition to that?

Nope. 

> I'm going to assign this to myself and give it a try at some point today.

Fantastic!

**Comment by MoOx** [comment]
Go! :)

**Comment by bjankord** [comment]
+1

**Comment by davidtheclark** [comment]
#172 adds this functionality.
